utils/frame/benchmarking-cli/src/lib.rs

```diff
@@ -109,34 +109,37 @@ impl BenchmarkCmd {
 		)
 		.execute(strategy.into())
 		.map_err(|e| format!("Error executing runtime benchmark: {:?}", e))?;
-		let results = <Option<Vec<BenchmarkResults>> as Decode>::decode(&mut &result[..])
-			.unwrap_or(None);
-
-		if let Some(results) = results {
-			// Print benchmark metadata
-			println!(
-				"Pallet: {:?}, Extrinsic: {:?}, Steps: {:?}, Repeat: {:?}",
-				self.pallet,
-				self.extrinsic,
-				self.steps,
-				self.repeat,
-			);
-
-			// Print the table header
-			results[0].0.iter().for_each(|param| print!("{:?},", param.0));
-
-			print!("extrinsic_time,storage_root_time\n");
-			// Print the values
-			results.iter().for_each(|result| {
-				let parameters = &result.0;
-				parameters.iter().for_each(|param| print!("{:?},", param.1));
-				// Print extrinsic time and storage root time
-				print!("{:?},{:?}\n", result.1, result.2);
-			});
-
-			eprintln!("Done.");
-		} else {
-			eprintln!("No Results.");
+		let results = <Result<Vec<BenchmarkResults>, Vec<u8>> as Decode>::decode(&mut &result[..])
```

Comment:
```suggestion
		let results = <Result<Vec<BenchmarkResults>, String> as Decode>::decode(&mut &result[..])
```

Reply:
I get `Error: "�Benchmark not found for this pallet."` if we don't decode twice. Other option is to encode the whole thing?, but maybe you don't want that, given your other comment of encoding just the error.

Reply:
I get `Error: "�Benchmark not found for this pallet."` if we don't decode twice. Other option is to encode the whole thing?, but maybe you don't want that, given your other comment of encoding just the error.

Reply:
Ahh yeah! Sorry!

Could you please use [`RuntimeString`](https://github.com/paritytech/substrate/blob/879e28ae38643c028fa09c157e7b2714bcbf02a1/primitives/runtime/src/runtime_string.rs#L23)

(You also need to change the `Benchmark` trait to directly return `Result<BenchmarkResult, RuntimeString>`

Reply:
Nice, thanks!